docs/guppi/index.rst

```diff
@@ -0,0 +1,132 @@
+rather than grouping the components of a :term:`complete sample` together.  For
+each channel, polarization samples from the same point in time are stored
+adjacent to one another.  At the end of each channel's data block is a section
+of **overlap samples** identical to the first samples in the next payload. 
+Baseband retains these redundant samples when reading individual GUPPI frames,
+but removes them when reading files as a stream.
+
+.. _guppi_usage:
+
+Usage
+=====
+
+This section covers reading and writing GUPPI files with Baseband; general
+usage is covered in the :ref:`Getting Started <getting_started>` section.  The
+examples below use the sample PUPPI file ``baseband/data/sample_puppi.raw``,
+and the the `astropy.units` and `baseband.guppi` modules::
+
+    >>> from baseband import guppi
+    >>> import astropy.units as u
+    >>> from baseband.data import SAMPLE_PUPPI
+
+Single files can be opened with `~baseband.guppi.open` in binary mode, which
+provides a normal file reader, but extended with methods to read a
+`~baseband.guppi.GUPPIFrame`::
+
+    >>> fb = guppi.open(SAMPLE_PUPPI, 'rb')
+    >>> frame = fb.read_frame()
+    >>> frame.shape
+    (8192, 2, 4)
+    >>> frame[:3, 0, 1]    # doctest: +FLOAT_CMP
+    array([14. +3.j, -1. -5.j, 30.-19.j], dtype=complex64)
+    >>> fb.close()
+
+Since the files can be quite large, the payload is mapped (with
+`numpy.memmap`), so that if one accesses part of the data, only the
+corresponding parts of the encoded payload are loaded into memory (since the
+sample file is encoded using 8 bits, the above example thus loads 6 bytes into
+memory).
+
+Opening in stream mode wraps the low-level routines such that reading and
+writing is in units of samples, and provides access to header information::
+
+    >>> fh = guppi.open(SAMPLE_PUPPI, 'rs')
+    >>> fh
+    <GUPPIStreamReader name=... offset=0
+        sample_rate=2500.0 Hz, samples_per_frame=7680,
+        sample_shape=SampleShape(npol=2, nchan=4), bps=8,
+        start_time=2018-01-14T14:11:33.000>
+    >>> d = fh.read()
+    >>> d.shape
+    (30720, 2, 4)
+    >>> d[:3, 0, 1]    # doctest: +FLOAT_CMP
+    array([14. +3.j, -1. -5.j, 30.-19.j], dtype=complex64)
+    >>> fh.close()
+
+Note that ``fh.samples_per_frame`` represents the number of samples per frame
+**excluding overlap samples**, since the stream reader works on a linearly
+increasing sequence of samples.  Frames themselves have access to the overlap,
+and ``fh.header0.samples_per_frame`` returns the number of samples per frame
+including overlap.
+
+To set up a file for writing as a stream is possible as well.  Overlap must be
+zero when writing (so we set ``samples_per_frame`` to its stream reader value
+from above)::
+
+    >>> from astropy.time import Time
+    >>> files = ['puppi_test.000{i}.raw'.format(i=i) for i in range(2)]
+    >>> fw = guppi.open(files, 'ws', frames_per_file=2, sample_rate=2500*u.Hz,
+    ...                 samples_per_frame=7680, pktsize=8192,
```

Comment:
how can `samples_per_frame` and `pktsize` be different for zero overlap?

Reply:
I haven't found any information on what `PKTSIZE` is supposed to represent.  I can set a default in `fromvalues`, though will still have to override it for the sample file because of how small it is.

Reply:
Okay, did a bit of digging, and `PKTSIZE` is an output packet size related to the FPGA architecture being used, so presumably it's related to `PKTFMT` being 1SFA.  I can't seem to find more info though...